include/zephyr/kernel_structs.h

```diff
@@ -258,8 +258,11 @@ extern atomic_t _cpus_active;
  */
 bool z_smp_cpu_mobile(void);
 
-#define _current_cpu ({ __ASSERT_NO_MSG(!z_smp_cpu_mobile()); \
```

Comment:
@andyross does preemption matters, if `arch_curr_cpu()` is just a single instruction?